### PR TITLE
[e2e] Simplify creation of VMIs in infra tests by using libvmi

### PR DESCRIPTION
**Pull request by brianmcarey**
**What this PR does / why we need it**:

Simplify creation of VMIs in infra tests by using libvmi

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [x] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [x] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [x] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [x] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [x] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Review by dhiller — COMMENTED** [review]
/approve

Thank you @brianmcarey

File: tests/infrastructure/prometheus.go [hunk 1]
```diff
@@ -260,17 +231,15 @@ var _ = DescribeInfra("[rfe_id:3187][crit:medium][vendor:cnv-qe@redhat.com][leve
...
-		tests.AppendEmptyDisk(vmi, "testdisk", v1.VirtIO, "1Gi")
-
+		vmi := libvmi.NewAlpine(libvmi.WithEmptyDisk("testdisk", v1.VirtIO, resource.MustParse("1G")))
 		if preferredNodeName != "" {
-			pinVMIOnNode(vmi, preferredNodeName)
-		}
-		nodeName := startVMI(vmi)
-		if preferredNodeName != "" {
-			Expect(nodeName).To(Equal(preferredNodeName), "Should run VMIs on the same node")
+			vmi = libvmi.NewAlpine(libvmi.WithEmptyDisk("testdisk", v1.VirtIO, resource.MustParse("1G")),
```
Comment (xpivarc) [inline tests/infrastructure/prometheus.go:236]: nit: what about constructing a slice of options and then passing it to `NewAlpine`?

**Review by xpivarc — COMMENTED** [review]
/approve
/lgtm
I note, that I think it would be much safer and globally correct if we did the following:
Rename function `NewRandomVMIWithEphemeralDisk` to be private.
Create `NewRandomVMIWithEphemeralDisk` function but implement it with libvmi.
Assert the previous function gives you the same vmi.
The last step replaces all calls with more specific libvmi variants.

**Pull request merged**